prowler/providers/microsoft365/services/entra/entra_thirdparty_integrated_apps_not_allowed/entra_thirdparty_integrated_apps_not_allowed.py

```diff
@@ -1,18 +1,40 @@
-from prowler.lib.check.models import Check, Check_Report_Microsoft365
+from typing import List
+
+from prowler.lib.check.models import Check, CheckReportMicrosoft365
 from prowler.providers.microsoft365.services.entra.entra_client import entra_client
 
 
 class entra_thirdparty_integrated_apps_not_allowed(Check):
-    def execute(self) -> Check_Report_Microsoft365:
-        findings = []
+    """Check if third-party integrated apps are not allowed for non-admin users in Entra.
+
+    This check verifies that non-admin users are not allowed to create third-party apps.
+    If the policy allows app creation, the check fails.
+
+    Attributes:
+        metadata: Metadata associated with the check (inherited from Check).
+    """
 
+    def execute(self) -> List[CheckReportMicrosoft365]:
+        """Execute the check to ensure third-party integrated apps are not allowed for non-admin users.
+
+        This method checks if the authorization policy allows non-admin users to create apps.
+        If the policy allows app creation, the check fails. Otherwise, the check passes.
+
+        Returns:
+            List[CheckReportMicrosoft365]: A list containing the result of the check for app creation policy.
+        """
+        findings = []
         auth_policy = entra_client.authorization_policy
-        report = Check_Report_Microsoft365(self.metadata(), auth_policy)
-        report.resource_name = getattr(auth_policy, "name", "Authorization Policy")
-        report.resource_id = getattr(auth_policy, "id", "authorizationPolicy")
+        report = CheckReportMicrosoft365(
+            metadata=self.metadata(),
+            resource=auth_policy,
+            resource_name=getattr(auth_policy, "name", "Authorization Policy"),
+            resource_id=getattr(auth_policy, "id", "authorizationPolicy"),
```

Comment:
```suggestion
            resource_name=auth_policy.name,
            resource_id=auth_policy.id,
```

Why are you using the `getattr` if the object has `name` and `id`?

Reply:
Mario designed it this way to prevent the check from crashing when the authentication policy couldn't be retrieved. If the access is directly made the test `Test_entra_thirdparty_integrated_apps_not_allowed` crashes instead of returning a `FAIL` finding.

Reply:
but as far as I can see these fields are present in the object created in the service, am I missing something?

Reply:
If the call to `msgraph` fails by any chance:


```python
auth_policy = await self.client.policies.authorization_policy.get()
```

 The object will be stored as `None` at the direct access will break the execution with `'NoneType' object has no attribute 'name'`.

It seems Mario implemented a safe retrieval flow to ensure that if we encounter issues fetching data from the API, the process doesn’t break. Instead, the tool returns a `FAIL` finding, due to the inability to properly assess the tenant's configuration.

Instead of that, we can craft a default object with non compliant configuration on purpose intead of `None` in the cases we can not retrieve the information, to standarize `CheckReportMicrosoft365` creation in all checks with direct access.

Reply:
That's interesting for sure, I did not realise of that part 🤔

Reply:
For me it is fine as it is or if we want it to be understood better I think this way is more readable, although `getattr` doesn't seem wrong in this case:

```
resource_name = auth_policy.name if auth_policy and auth_policy.name else “Authorization policy”.
resource_id = auth_policy.id if auth_policy and auth_policy.id else “authorizationPolicy”
```

Another option is to make a refactor so that the object that is returned always has all the fields (good practice), in case of not being able to dedicate time to this I would stay with the option above

Reply:
I have implemented a new output for `report.status_extended` to let the user know exactly what is happening in addition to your comments @AdriiiPRodri. Thanks for your help!